### DISCUSSION TITLE
About sweeping...

### DISCUSSION THREAD

**Discussion by almostSouji** [discordjs/discord.js#5247] [General]
Cache cleanup is generally user duty, as it can have major side effects on events and base library functionality. Yet we currently have native options and handling for message sweeping, which can be faulty as well: #4599. 

I'm still uncertain if this is a thing we should fix and make it more reliable or remove native sweeping completely and let the developer do it on their own accord.

**Comment by MattIPv4** [comment]
For me, given caching is currently a built-in part of the library, I think it makes sense to also have some sweeping of that be built-in. The folks writing the library and the logic that interacts with the caching system are likely also going to be the best folks for knowing how to safely sweep it.

If caching gets pulled out of the core lib though, and becomes something the user has far more control over (able to provide their own caching system), then I'd definitely fall far more into the boat of removing sweeping from the lib.
